openwisp_monitoring/device/writer.py

```diff
@@ -0,0 +1,398 @@
+import logging
+from copy import deepcopy
+from datetime import datetime, timedelta
+
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ObjectDoesNotExist
+from pytz import UTC
+from swapper import load_model
+
+from .. import settings as monitoring_settings
+from ..monitoring.configuration import ACCESS_TECHNOLOGIES
+
+Chart = load_model('monitoring', 'Chart')
+Metric = load_model('monitoring', 'Metric')
+AlertSettings = load_model('monitoring', 'AlertSettings')
+Device = load_model('config', 'Device')
+
+logger = logging.getLogger(__name__)
+
+
+class DeviceDataWriter(object):
```

Comment:
Let's add a description here. Something like:

This class is in charge of writing the device metric data.
Before these methods were shipped in the REST API view
but later have been moved here to allow writing this data in
the background processes of OpenWISP.